core/domain/user_services.py

```diff
@@ -499,6 +501,161 @@ def get_user_settings_by_auth_id(
         return None
 
 
+def get_all_user_groups() -> List[user_domain.UserGroup]:
+    """Return the list of user groups.
+
+    Returns:
+        List[user_domain.UserGroup]. List of all user groups.
+    """
+    user_group_models: List[user_models.UserGroupModel] = list(
+        user_models.UserGroupModel.get_all())
+    user_groups_list = []
+    for user_group_model in user_group_models:
+        user_usernames: List[str] = []
+        for user_setting in get_users_settings(user_group_model.user_ids):
+            assert user_setting is not None
+            assert user_setting.username is not None
+            user_usernames.append(user_setting.username)
+        user_group = user_domain.UserGroup(
+            user_group_model.id, user_group_model.name, user_usernames)
+        user_groups_list.append(user_group)
+    return user_groups_list
+
+
+def _check_if_usernames_are_valid(
+    user_group_name: str, user_group_user_usernames: List[str]
+) -> None:
+    """Checks if the given list of users are valid or not.
+
+    Args:
+        user_group_name: str. The name of the user group.
+        user_group_user_usernames: List[str]. The list of usernames for
+            which validation needs to be done.
+
+    Raises:
+        Exception. The user inside user group does not exist.
```

Comment:
Your function has two types of exceptions, but this only describes one.

Reply:
Updated